### PR TITLE
[FLINK-25565][Formats][Parquet] write and read parquet int64 timestamp

### PR DESCRIPTION
**Pull request by cuibo01**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by cuibo01** [comment]
@rmetzger @twalthr @AHeise @zentol can you help me review this PR... thx

**Comment by cuibo01** [comment]
@MartijnVisser @tillrohrmann can you help me review this PR... thx

**Comment by MartijnVisser** [comment]
@cuibo01 Thank you for your contribution. Please don't ping multiple people. It can be quite annoying for people who are working on open source contributions and can actually cause people to ignore these types of pings. As you can see (based on the number of open PRs), there are a lot of open items that need attention. Everyone tries their best to get as many contributions in as possible. The better the PR is, the higher chance it can be merged quickly.

Since we are approaching the release 1.15 deadline, I'm not sure this can be reviewed quickly.

**Comment by cuibo01** [comment]
> @cuibo01 Thank you for your contribution. Please don't ping multiple people. It can be quite annoying for people who are working on open source contributions and can actually cause people to ignore these types of pings. As you can see (based on the number of open PRs), there are a lot of open items that need attention. Everyone tries their best to get as many contributions in as possible. The better the PR is, the higher chance it can be merged quickly.
> 
> Since we are approaching the release 1.15 deadline, I'm not sure this can be reviewed quickly.

ok thx @MartijnVisser

**Comment by cuibo01** [comment]
> @cuibo01 Thank you for your contribution. Please don't ping multiple people. It can be quite annoying for people who are working on open source contributions and can actually cause people to ignore these types of pings. As you can see (based on the number of open PRs), there are a lot of open items that need attention. Everyone tries their best to get as many contributions in as possible. The better the PR is, the higher chance it can be merged quickly.
> 
> Since we are approaching the release 1.15 deadline, I'm not sure this can be reviewed quickly.

hi @MartijnVisser ，when is the release 1.15 deadline ?

**Comment by MartijnVisser** [comment]
> hi @MartijnVisser ，when is the release 1.15 deadline ?

February 4th

**Comment by cuibo01** [comment]
> > hi @MartijnVisser ，when is the release 1.15 deadline ?
> 
> February 4th

Thank you very much.

File: flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetFileFormatFactory.java [hunk 1]
```diff
@@ -62,6 +62,21 @@
                                     + " time and LocalDateTime. Hive 0.x/1.x/2.x use local timezone. But Hive 3.x"
                                     + " use UTC timezone");
 
+    public static final ConfigOption<String> TIMESTAMP_TIME_UNIT =
+            key("timestamp.time.unit")
+                    .stringType()
+                    .defaultValue("millis")
```
[the remark this answers is not in the document]
Comment (cuibo01) [reply flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetFileFormatFactory.java]: Updated

File: flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetFileFormatFactory.java [hunk 2]
```diff
@@ -62,6 +62,21 @@
                                     + " time and LocalDateTime. Hive 0.x/1.x/2.x use local timezone. But Hive 3.x"
                                     + " use UTC timezone");
 
+    public static final ConfigOption<String> TIMESTAMP_TIME_UNIT =
```
[the remark this answers is not in the document]
Comment (cuibo01) [reply flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetFileFormatFactory.java:65]: yes, they are the same, and flink add enumerated values 'value is nanos/micros/millis'

[the remark this answers is not in the document]
Comment (cuibo01) [reply flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetFileFormatFactory.java:65]: in flink-parquet,  flink adds and deletes 'parquet.' prefixes by itself.

**Comment by cuibo01** [comment]
> > hi @MartijnVisser ，when is the release 1.15 deadline ?
> 
> February 4th

hi @MartijnVisser when will the 1.15 be released? thx

**Comment by MartijnVisser** [comment]
> > > hi @MartijnVisser ，when is the release 1.15 deadline ?
> > 
> > 
> > February 4th
> 
> hi @MartijnVisser when will the 1.15 be released? thx

Hi @cuibo01 - The first release candidate of 1.15 has been released this week. I expect that in the next 2 weeks it will be officially released.

**Comment by MartijnVisser** [comment]
@cuibo01 Can you please rebase and squash your commits?

**Comment by MartijnVisser** [comment]
@cuibo01 Also don't forget to run `mvn spotless:apply` since the CI still fails for your PR due to spotless errors

**Comment by cuibo01** [comment]
> @cuibo01 Thanks for the PR; I do think this would also require some documentation updates, can you have a look at that?

thx @MartijnVisser , i have updated `parquet.md`

**Comment by tweise** [comment]
@cuibo01 @MartijnVisser I just run into this issue (Flink 1.17) and was happy to find this PR. Any reason this stalled? Can I help get this to the finish line?

**Comment by cuibo01** [comment]
@tweise , Of course you can, thx

**Comment by tweise** [comment]
@cuibo01 thanks, I resolved conflicts and pushed the merge for testing - will do a rebase after we see tests pass.

**Comment by tweise** [comment]
@flinkbot run azure

**Comment by tweise** [comment]
@flinkbot run azure

**Comment by tweise** [comment]
@flinkbot run azure

**Comment by tweise** [comment]
Also verified with my application.

**Pull request merged**